application/backend/src/main/resources/apps/typerefinery/components/graphs/barchart/clientlibs/functions.js

```diff
@@ -156,10 +156,36 @@ const DEFAULT_BAR_CHART_DATA = {
             // Passing {} because, The values from the model obj are fetched in bellow function definition.
             ns.updateComponentHTML({}, $component);
         }
-
+        ns.updateChartInstance = (data, $component) => {
+          if (!$component) {
+            console.log('[Barchart/clientlibs/functions.js] component does not exist')
+            return;
+          }
+          let componentConfig = componentNs.getComponentConfig($component);
+          componentConfig = {
+            ...componentConfig,
+            ...DEFAULT_BAR_CHART_DATA
+          }
+    
+          graphItemsNs[componentConfig.resourcePath].data = {
+              labels: data.labels || componentConfig.labels,
+              datasets: [
+                {
+                  axis: "y",
+                  data: data.chartData || componentConfig.chartData,
+                  fill: false,
+                  backgroundColor:"blue",
```

Comment:
Please check properly and test.

**Dont give static value**

Reply:
Done